packages/react-hooks/src/hooks/media.ts

```diff
@@ -4,7 +4,13 @@ type EffectHook = typeof useEffect | typeof useLayoutEffect;
 
 function createUseMediaFactory(useEffectHook: EffectHook) {
   return (query: string) => {
-    const [match, setMatch] = useState(false);
+    const [match, setMatch] = useState(() => {
+      if (typeof window === 'undefined' || !window.matchMedia) {
```

Comment:
The useEffectHook below seems to be doing a couple more things than what was added in here (mainly adding the event listener). Do you think those steps should be added in here as well (and potentially extract the code somewhere so we don't have code duplication)?

Reply:
React lazy state initialization should not have any side effects such as adding event listeners, so we definitely don't want to pull that in here.

I'm not sure what we could extract to remove any duplication. Do you have an example of what you're thinking? Only thing I could see is something like this, but IMO this is much more difficult to reason about. Maybe you have another solution in mind?

```tsx
const getMatchMedia = useCallback((query: string) => {
  if (typeof window === 'undefined' || !window.matchMedia) {
    return;
  }
  return window.matchMedia(query);
}, []);

const [match, setMatch] = useState(
  () => getMatchMedia(query)?.matches ?? false,
);
useEffectHook(() => {
  const matchMedia = getMatchMedia(query);
  if (!matchMedia) {
    return;
  }
  const updateMatch = (event: MediaQueryListEvent) =>
    setMatch(event.matches);
  setMatch(matchMedia.matches);
  matchMedia.addListener(updateMatch);
  return () => {
    matchMedia.removeListener(updateMatch);
  };
}, [getMatchMedia, query]);

return match;
```